java-checks/src/test/files/checks/SynchronizedLockCheck.java

```diff
@@ -0,0 +1,48 @@
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.Lock;
+
+class A {
+  void foo() {
+    Lock lock = new MyLockImpl();
+    synchronized (lock) { // Noncompliant
+    }
+    synchronized (new MyLockImpl()) { // Noncompliant
+    }
+    synchronized (this) { // Compliant
+    }
+    synchronized (bar()) { // Compliant
```

Comment:
Unecessary, you are just testing type expression resolution with this, not your actual rule.